ts/models/conversation.ts

```diff
@@ -709,6 +710,130 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
       return null;
     }
   }
+
+
+  // public sendTypingMessage(isTyping: boolean) {
+  //   if (!this.isPrivate()) {
+  //     return;
+  //   }
+
+  //   const recipientId = this.id;
+
+  //   if (!recipientId) {
+  //     throw new Error('Need to provide either recipientId');
+  //   }
+
+  //   const primaryDevicePubkey = window.storage.get('primaryDevicePubKey');
+  //   if (recipientId && primaryDevicePubkey === recipientId) {
+  //     // note to self
+  //     return;
+  //   }
+
+  //   const typingParams = {
+  //     timestamp: Date.now(),
+  //     isTyping,
+  //     typingTimestamp: Date.now(),
+  //   };
+  //   const typingMessage = new TypingMessage(typingParams);
+
+  //   // send the message to a single recipient if this is a session chat
+  //   const device = new PubKey(recipientId);
+  //   getMessageQueue()
+  //     .sendToPubKey(device, typingMessage)
+  //     .catch(window?.log?.error);
+  // }
+
+  /**
+   * Creates an unsend request using protobuf and adds to messageQueue.
+   * @param message Message to unsend
+   * @returns 
+   */
+  public unsendMessage(message: MessageModel) {
+    // TODO: remove logging
+    window.log.info('Sending unsend request');
+    console.log({ destination: this.id });
+
+    //#region checking for early exit conditions
+    if (!message.getPropsForMessage().messageHash) {
+      console.log("@ Message has no hash - aborting unsend message");
+      return;
+    }
+
+    // handling is private (idk what that means)
+    if (!this.isPrivate()) {
+      console.log({ isPrivate: this.isPrivate });
+      return;
+    }
+
+    // handling no destination
+    const destinationId = this.id;
+    if (!destinationId) {
+      return;
+    }
+
+    const ownPrimaryDevicePubkey = window.storage.get('primaryDevicePubKey');
+    if (destinationId && ownPrimaryDevicePubkey === destinationId) {
+      // note to self
+      return;
+    }
+    //#endregion
+
+    //#region building request
+    const author = message.isOutgoing() ? ownPrimaryDevicePubkey : message.id;
```

Comment:
this looks incorrect.
message.id is not a public key. You sure you are not looking for message.get('source') ?